core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts

```diff
@@ -1,23 +1,43 @@
-import { CommonModule } from '@angular/common';
-import { Component, OnInit, inject } from '@angular/core';
+import { Subject, fromEvent } from 'rxjs';
+
+import { AsyncPipe, NgFor, NgIf } from '@angular/common';
+import { Component, ElementRef, Inject, OnDestroy, OnInit, ViewChild, inject } from '@angular/core';
 import { FormsModule } from '@angular/forms';
 import { ActivatedRoute, Params, Router } from '@angular/router';
 
-import { SafeUrlPipe } from '@dotcms/ui';
+import { DialogModule } from 'primeng/dialog';
+
+import { takeUntil } from 'rxjs/operators';
+
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+import { DotSpinnerModule, SafeUrlPipe } from '@dotcms/ui';
 
 import { EditEmaStore } from './store/dot-ema.store';
 
 import { DotPageApiService } from '../services/dot-page-api.service';
+import { NG_CUSTOM_EVENTS, POST_MESSAGE_ACTIONS, WINDOW } from '../shared/models';
 
 @Component({
     selector: 'dot-ema',
     standalone: true,
-    imports: [CommonModule, FormsModule, SafeUrlPipe],
-    providers: [EditEmaStore, DotPageApiService],
+    imports: [NgFor, NgIf, AsyncPipe, FormsModule, SafeUrlPipe, DialogModule, DotSpinnerModule],
+    providers: [
+        EditEmaStore,
+        DotPageApiService,
+        {
+            provide: WINDOW,
+            useValue: window
+        }
+    ],
```

Comment:
Does this component not use `OnPush` for any reason?

Reply:
Added!